### PR TITLE
127 video overlay in hero

### LINKED ISSUE #127: Video overlay in Hero
Like in https://www.moleculardevices.com/products/microplate-readers/multi-mode-readers

### PR DESCRIPTION
**Pull request by paolomoz**
Fix #127

Test URLs:
- Before: https://main--moleculardevices--hlxsites.hlx.live/products/axon-patch-clamp-system/digitizers/axon-digidata-1550b-plus-humsilencer
- After: https://127-video-overlay-in-hero--moleculardevices--hlxsites.hlx.live/products/axon-patch-clamp-system/digitizers/axon-digidata-1550b-plus-humsilencer

### CODE REVIEW

**Comment by paolomoz** [comment]
> The video should auto play once open in the lightbox.

set autoplay if browser does not block
https://knowledge.vidyard.com/hc/en-us/articles/360009879754-Use-query-parameters-and-data-attributes-to-override-player-settings#:~:text=Notes-,autoplay,-0%2C%201%20or

File: blocks/hero/hero.js [hunk 1]
```diff
@@ -1,14 +1,92 @@
...
+function getUrls(text) {
+  const match = text.match(/\b((https?|ftp|file):\/\/|(www|ftp)\.)[-A-Z0-9+&@#/%?=~_|$!:,.;]*[A-Z0-9+&@#/%=~_|$]/ig);
+  if (match) return match;
+  return null;
+}
+
+function getVideoUrl(text) {
+  let videoUrl = null;
+  if (getUrls(text)) {
+    const hostnames = ['vids.moleculardevices.com', 'share.vidyard.com'];
```
Comment (lydiapuric) [inline blocks/hero/hero.js:23]: I would see the functionality for video handling more centralized incl. it's validation.

Comment (paolomoz) [reply blocks/hero/hero.js]: > I would see the functionality for video handling more centralized incl. it's validation.

I moved the video handling functions in a separate js as part of the video block.

**Comment by paolomoz** [comment]
> The video should auto play once open in the lightbox.

done

**Review by lydiapuric — APPROVED** [review]
LGTM +1

**Pull request merged**